docs/root/version_history/current.rst

```diff
@@ -74,6 +74,7 @@ New Features
 * http: added support for :ref:`%DOWNSTREAM_PEER_FINGERPRINT_1% <config_http_conn_man_headers_custom_request_headers>` as custom header.
 * http: added :ref:`allow_chunked_length <envoy_v3_api_field_config.core.v3.Http1ProtocolOptions.allow_chunked_length>` configuration option for HTTP/1 codec to allow processing requests/responses with both Content-Length and Transfer-Encoding: chunked headers. If such message is served and option is enabled - per RFC Content-Length is ignored and removed.
 * http: introduced new HTTP/1 and HTTP/2 codec implementations that will remove the use of exceptions for control flow due to high risk factors and instead use error statuses. The old behavior is used by default, but the new codecs can be enabled for testing by setting the runtime feature `envoy.reloadable_features.new_codec_behavior` to true. The new codecs will be in development for one month, and then enabled by default while the old codecs are deprecated.
+* http: modified the HTTP header-map data-structure to use an underlying dictionary and a list (no change to the header-map API). By default the use of a dictionary is disabled. It can be enabled by setting the `envoy.http.headermap.lazy_map_min_size` runtime feature to a non-negative number which defines the minimal number of headers in a request/response/trailers required for using a dictionary in addition to the list.
```

Comment:
Presumably eventually we want to actually configure this by default to some value? Should we rephrase slightly that this is in place for initial testing, and possibly recommend a threshold? Then we will follow up later with a change to lock in a default once we have done some testing?

Reply:
I've updated the description, adding a suggested value.
I agree that once we find a good threshold (after more testing in real world workloads), we can set it and notify about it in future versions.